### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by VibhuJawa**
This PR enables GPU CI on this repo: 

Tested locally using: 
```
docker run --gpus all -v $(pwd):/workspace/crossfit --rm -it rapidsai/base:24.06-cuda12.2-py3.11  bash -c "cd /workspace/crossfit && bash ci/test_gpu.sh"
```

### CODE REVIEW

**Comment by VibhuJawa** [comment]
CI seems to be passing again . 

CC: @marcromeyn / @edknv  for quick eyes on this.

File: crossfit/backend/cudf/array.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def __init__(self):
         def concatenate(self, series_list, *, axis=None):
             return cudf.concat(series_list, axis=axis or 0)
 
-    np_backend_dispatch.register((cudf.Series, cudf.GenericIndex))(CudfBackend())
+    np_backend_dispatch.register((cudf.Series, cudf.Index))(CudfBackend())
```
Comment (VibhuJawa) [inline crossfit/backend/cudf/array.py:32]: Unsure if this is the right fix. CC: @rjzamora

**Review by sarahyurick — APPROVED** [review]
LGTM! Can also confirm this works fine with my NeMo Curator setup.

File: crossfit/backend/cudf/array.py (hunk 1 above)
Comment (rjzamora) [reply crossfit/backend/cudf/array.py:32]: Yes, this looks correct to me. My understanding is that all "real" index objects inherit from `Index` now.

I suppose dask-cudf uses [BaseIndex](https://github.com/rapidsai/cudf/blob/34dea6fe40fc20966b48257853865111df4a687f/python/dask_cudf/dask_cudf/backends.py#L52), but that dispatch was added a long time ago (and BaseIndex is probably unnecessary).

**Pull request merged**